upt_macports/templates/perl.Portfile

```diff
@@ -5,16 +5,40 @@ PortGroup           perl5 1.0
 {% endblock %}
 {% block nameversion %}
 perl5.branches      5.26 5.28
-perl5.setup         {{ pkg._pkgname() }} {{ pkg.upt_pkg.version }}
+perl5.setup         {{ pkg._pkgname() }} {{ pkg.upt_pkg.version }}{{ pkg._cpandir() }}
 revision            0
 {% endblock %}
 
 {% block versions %}
+{%- if pkg.upt_pkg.requirements.run or pkg.upt_pkg.requirements.test or pkg.upt_pkg.requirements.build %}
 if {${perl5.major} != ""} {
+    {%- if pkg.upt_pkg.requirements.build|length > 0 %}
+
+    depends_build-append \
+                    {% for i in pkg.upt_pkg.requirements.build|sort(attribute='name')|unique(attribute='name') %}
+                    port:p${perl5.major}-{{i.name|replace('::','-')|lower}}{% if not loop.last %} \
```

Comment:
Why not use "_normalized_macports_name" here, since it is a static method? Should "lower" be part of this method as well?

Reply:
I don't think we can keep lower as part of "_normalized_macports_name" as for "perl5.setup" we need the normal version.

Reply:
Yes, there is indeed the need for a different name for the ```perl5.setup``` line and the dependencies list... I am sure there are different options, but what it's right now seems fine to me. It doesn't really hurt the readability of the template, so I would just leave it for now in the interest of merging this soon...

Reply:
My personal preference would be to call a function rather than regexping on spot, in particular for cases when suddenly one decides to change the convention (like change capitalisation), but I leave it to the mentors.

I would in fact use `pkg.whatever_the_function_name_is(i.name)` to replace the full `port:p${perl5.major}-{{i.name|replace('::','-')|lower}}`.

Reply:
as I said that is a possibility, it would need another function like ```pkg._macports_dependency_name``` that does an additional ```.lower()``` on ```pkg._normalized_macports_name```

Reply:
@mojca: for now, we can probably keep it like that. But in the future, we could move this whole section back to the base template, and have a custom filter do the "formatting" of the package name.

Reply:
I *think* I see now what @mojca means: we should add a function in the backend(s) that has ```name``` as argument and returns the complete dependency line. Something like (as pseudo-code):

```
@staticmethod
def _macports_dependency(name):
    dep_type = 'p${perl5.major}'
    dep_name = pkg._normalized_macports_name(name).lower()
    return f'port:{dep_type}-{dep_name}'
```

I do like that suggestion and we should probably do that for every "frontend" that is available. @Korusuke can you please add that to your to-do list (it can probably go into one PR with separate commits for the different Python/Perl/Ruby frontends)?

```

Reply:
Yeah sure, that same function can also decide the naming for example for python ports if they are already there on MacPorts. It can also check for if ports are already present or not. 
We can implement this during recursive feature addition.

Reply:
Let's do that later :)